src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlConverter.cs

```diff
@@ -1082,45 +1086,63 @@ public static int ToChars(DateTime value, byte[] chars, int offset)
             return offset - offsetMin;
         }
 
-        public static bool IsWhitespace(string s)
+        public static bool IsWhitespace(ReadOnlySpan<char> chars) =>
+            chars.IndexOfAnyExcept(s_whitespaceChars) < 0;
+
+        public static bool IsWhitespace(ReadOnlySpan<byte> bytes) =>
+            bytes.IndexOfAnyExcept(s_whitespaceBytes) < 0;
+
+        public static bool IsWhitespace(char ch) =>
+            ch is <= ' ' and (' ' or '\t' or '\r' or '\n');
```

Comment:
Is `char.IsWhitespace(char)` not an option here?
If it's too slow it should be made faster, or an extra internal method for fast-path compiroson added, to avoid sprinkling around different implementations of whitespace-check (although code is trivial, we should use existing APIs).

Reply:
Sadly, every single standard and implementation has a different idea of what `WhiteSpace` is.

In this case, `char.IsWhiteSpace` also includes a bunch of non-Ascii chars.

Reply:
Hooray for standards 😄 

A bit-mask approach here isn't worth it? (as you've done e.g. in Markdig)

Reply:
I'm very familiar with Markdig and a complete stranger to XML. I also made the decision to not care about 32-bit in Markdig 😄 

I assume that a bunch of these checks indeed _could_ use better implementations (like the one for [HexConverter.IsHexChar](https://github.com/dotnet/runtime/blob/ea4f2cc6190b1fbcc53e27e9b21a91eeea5b7736/src/libraries/Common/src/System/HexConverter.cs#L309-L338)), but that adds a bit more code to maintain and isn't specific to just Xml/this PR.

Reply:
Also, something about `System.Private.Xml` makes VS absolutely destroy all the memory and CPU and brings my machine to a crawl, and I don't find notepad coding quite as enjoyable :)

Reply:
> Also, something about System.Private.Xml makes VS absolutely destroy all the memory and CPU and brings my machine to a crawl, and I don't find notepad coding quite as enjoyable :)

cc: @sharwell